cep-XXXX.md

```diff
@@ -0,0 +1,154 @@
+
+3. If the DSO was in `C:\Windows\System32`, then these are system libraries and are
+   allowed.
+
+4. If the DSO was in the `DEFAULT_WIN_WHITELIST` location defined by the build tool,
+   then these are system libraries and are allowed. `DEFAULT_WIN_WHITELIST` is a
+   hardcoded constant in `conda-build` and `rattler-build`.
+
+Note that for Linux and macOS, the user has a way of controlling which system
+DSOs are allowed. On Linux, a user can modify the `sysroot_linux-*` package
+and on macOS, the user can specify a custom macOS SDK using `CONDA_BUILD_SYSROOT`.
+However, the user has no way of controlling which system DSOs are allowed
+on Windows.
+
+### Need for customizability of allowed Windows system libraries
+
+A user of conda build tools might need to customize the set of system DLLs.
+An example from Linux is the [removal of `libnsl` from `glibc`](https://github.com/conda-forge/linux-sysroot-feedstock/pull/40).
+
+In this case, newer versions of `glibc` dropped `libnsl` and made it a separate package,
+therefore conda-forge dropped it from `sysroot_linux-*` packages to make sure
+when new conda packages link to the `libnsl` system library, the build tool will
+warn or error out.
+
+An example for Windows can be found in this issue:
+[Warn when linking to debug VC runtimes](https://github.com/conda/conda-build/issues/5732).
+
+In this case, the CI machines provide debug VC (Visual Studio) runtime DLLs and are
+considered system DLLs by conda build tools, but these debug DLLs
+are not always found in user systems. Also, linking to these debug DLLs
+are not desirable and therefore should not be shipped it the corresponding
+conda package (e.g. `vc14_runtime`). Therefore we need a way to tell the build tool
+which DLLs are allowed and which are not.
+
+## Specification
+
+This CEP proposes a mechanism to control the allowed system DLLs on Windows.
+When a JSON file provided by a conda package in `host` or `build` exists in
+any of the following locations we use those list of files:
+
+- `<BUILD_PREFIX>/etc/conda-build/dsolists.d/<conda-package-name>.json`
+
+- `<PREFIX>/etc/conda-build/dsolists.d/<conda-package-name>.json`
+
+The JSON file format is:
+
+```json
+{
+  "version": 1,
+  "allow": [str],
+  "deny": [str],
+  "subdir": str
+}
+```
+
+Here `version` is for the version of the DSO list JSON schema and will always
+be present for these files. The build tools should check the version and
+validate that the `version` is a known and supported version and error out
+if not.
+
+The elements of the two lists `allow` and `deny` supports standard POSIX
+globbing syntax; i.e., the following is allowed:
+
+```json
+{
+  "version": 1,
+  "allow": [
+    "C:/Windows/System32/*.dll",
+    "**/R.dll",
+  ],
+  "deny": [
+    "**/ucrtbased.dll",
+  ],
+  "subdir": "win-64",
+}
+```
+
+Note that forward slashes are used for path separation and only absolute paths
+are allowed.
```

Comment:
Are these rules (only absolute paths, standard POSIX globbing syntax) supposed to be checked by the build tools?